src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -88,9 +92,130 @@ public void oldSchoolWay() {
                     extor.shutdownNow();
                 }
             } catch (final InterruptedException exp) {
-                Thread.currentThread().interrupt();
                 extor.shutdownNow();
             }
         }
     }
+
+    /**
+     * Execute the tasks concurrently using {@link Threads} when
+     *  {@link ExecutorService} was initiated by someone else.
+     */
+    @Test
+    public void cactoosWay() {
+        this.repeatAtLeastTenTimesWithTimeoutInFiveSeconds(
+            arg -> {
+                final ExecutorService extor = Executors.newFixedThreadPool(3);
+                try {
+                    MatcherAssert.assertThat(
+                        new Threads<String>(
+                            extor,
+                            () -> {
+                                TimeUnit.SECONDS.sleep(3);
+                                return "txt 1";
+                            },
+                            () -> {
+                                TimeUnit.SECONDS.sleep(3);
+                                return "txt 2";
+                            },
+                            () -> {
+                                TimeUnit.SECONDS.sleep(3);
+                                return "txt 3";
+                            }
+                        ),
+                        new HasValues<>("txt 1", "txt 2", "txt 3")
+                    );
+                } finally {
+                    try {
+                        extor.shutdown();
+                        if (!extor.awaitTermination(1, TimeUnit.SECONDS)) {
+                            extor.shutdownNow();
+                        }
+                    } catch (final InterruptedException exp) {
+                        extor.shutdownNow();
+                    }
+                }
+            }
+        );
+    }
+
+    /**
+     * Execute the test at least 10 times with timeout in 5 seconds each.
+     * @param test The test to execute.
+     * @todo #972:30min Move this method to new object RepeatWithTimeout.
+     *  This object might be present in cactoos/src/test/java/o.c.experimental
+     *  in order to be used with concurrent unit tests.
+     */
+    public void repeatAtLeastTenTimesWithTimeoutInFiveSeconds(
+        final Proc<Boolean> test
```

Comment:
@dgroup I fail to see why this Proc must accept a boolean? Am I missing something?

Reply:
@llorllale, the `Proc` has a contract 
```java
public interface Proc<X> {
    /**
     * Execute it.
     * @param input The argument
     * @throws Exception If fails
     */
    void exec(X input) throws Exception;
```
we can change to object or whatever.
This argument wasn't used within the test.

We can fix it using the following way
```java
public final class ThreadsTest {
    ...
    @Test
    public void cactoosWay() {
        this.repeatWithTimeout(
            () -> {
                final ExecutorService extor = Executors.newFixedThreadPool(3);
                try {
                    MatcherAssert.assertThat(
                        new Threads<String>(
                            extor,
                            () -> {
                                TimeUnit.SECONDS.sleep(3);
                                return "txt 1";
                            },
                            () -> {
                                TimeUnit.SECONDS.sleep(3);
                                return "txt 2";
                            },
                            () -> {
                                TimeUnit.SECONDS.sleep(3);
                                return "txt 3";
                            }
                        ),
                        new HasValues<>("txt 1", "txt 2", "txt 3")
                    );
                } finally {
                    try {
                        extor.shutdown();
                        if (!extor.awaitTermination(1, TimeUnit.SECONDS)) {
                            extor.shutdownNow();
                        }
                    } catch (final InterruptedException exp) {
                        extor.shutdownNow();
                    }
                }
            }
        );
    ...
    /**
     * Execute the test at least 10 times with timeout in 5 seconds each.
     * @param test The test to execute.
     */
    private void repeatWithTimeout(final SingleTest test) {
        MatcherAssert.assertThat(
            new UncheckedFunc<>(
                new Repeated<>(
                    new TimedFunc<Boolean, Boolean>(
                        input -> {
                            test.test();
                            return true;
                        },
                        Duration.ofSeconds(5).toMillis()
                    ),
                    10
                )
            ).apply(true),
            new IsEqual<>(true)
        );
    }

    private interface SingleTest {
        void test();
    }
}
```

Reply:
@dgroup you can just declare it as `Proc<?>`

Reply:
We can, but then we will need to pass a `null` to `test.exec`
```java
new TimedFunc<Boolean, Boolean>(
    input -> {
        test.exec(null);
        return true;
    },
    Duration.ofSeconds(5).toMillis()
)
```

Reply:
@dgroup I don't see why, considering that the generic wildcard will basically be interpreted as `java.lang.Object`.

Reply:
@llorllale 
![image](https://user-images.githubusercontent.com/1651114/50644819-f7992d00-0f97-11e9-9140-489f8d594968.png)

Reply:
@dgroup the issue is that `input` is a boolean because it's an input to `TimedFunc<Boolean, Boolean>`. Just pass `null` - should work.

Reply:
Ok, raising.